### PR TITLE
NMS-12529: Split OpenNMS and Minion docker setup

### PR DESCRIPTION
**Pull request by indigo423**
### All Contributors

* [x] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [x] Have you made a comment in that issue which points back to this PR?
* [x] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [x] If this is a new or updated feature, is there documentation for the new behavior?
* [ ] If this is new code, are there unit and/or integration tests?
* [ ] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

Instead of having one large Docker setup, I've split up the [Docker run instructions](http://docs.opennms.org/opennms/releases/25.2.0/guide-install/guide-install.html#_run_with_docker) for each component to be more consistent. It provides a docker-compose file for each component, such as Horizon core instance and a Minion which is more realistic to real-world setups.

### Reviewer Hint

We currently have no example of how to run a Sentinel in Docker. @agalue maybe you can help us out here.

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-12529

### CODE REVIEW

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 1]
```diff
@@ -1,38 +1,37 @@
 
-== Run with Docker
+=== Run with Docker
 
 Modern infrastructure allows you to deploy and run workloads in containers.
 With _{opennms-product-name}_ we provide and publish container images on link:https://hub.docker.com/u/opennms[DockerHub].
 
 NOTE: We don't install all available plugins in our published Docker image.
       If you want ot customize and maintain your own image, you can find the _Dockerfiles_ in our link:https://github.com/OpenNMS/opennms/tree/develop/opennms-container[source repository].
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
      If you want to customize and maintain your own image, you can find the _Dockerfiles_ in our link:https://github.com/OpenNMS/opennms/tree/develop/opennms-container[source repository].
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 2]
```diff
@@ -1,38 +1,37 @@
...
 
 * Run _{opennms-product-name}_ using _Docker Compose_ with a basic setup and _PostgreSQL_ on your local system as a quickstart
 * Persist RRD files from _{opennms-product-name}_ and _PostgreSQL_ in a volume
-* Run and configure a _Minion_ in the stack and connect it to the _{opennms-product-name}_ instance using environment variables
 * Introduce a reference with all available configuration and mount conventions for more advanced setups
 
-=== Before you begin
+==== Before you begin
 
 It is required you have at least the following components installed:
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
You must have at least the following components installed:
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 3]
```diff
@@ -1,38 +1,37 @@
...
 
 * Current stable _Docker_ release installed, e.g. installed from link:https://docs.docker.com/[Docker Documentation]
 * Current stable _Docker Compose_ installed, e.g. installed from link:https://docs.docker.com/compose/install/[Docker Compose instructions]
 * You should have a basic knowledge about _Docker_, _Docker Compose_ with networking, persisting files and mounting directories
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
* You should have a basic knowledge of _Docker_, _Docker Compose_ with networking, persisting files and mounting directories
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 4]
```diff
@@ -1,38 +1,37 @@
...
-=== Quickstart service stack
+==== Quickstart service stack
 
 // No section numbers for step-by-step guide
 :!sectnums:
 
-==== Step 1: Create service stack for PostgreSQL and _{opennms-product-name}_
+===== Step 1: Create service stack for PostgreSQL and _{opennms-product-name}_
 
 The first section describes how to setup _{opennms-product-name}_ service stack in a `docker-compose.yml` file.
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
The first section describes how to set up _{opennms-product-name}_ service stack in a `docker-compose.yml` file.
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 5]
```diff
@@ -1,38 +1,37 @@
...
 Create a project directory withg `mkdir opennms-horizon` and create inside a `docker-compose.yml` file with the following content:
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
Create a project directory with `mkdir opennms-horizon` and create a `docker-compose.yml` file in that directory with the following content:
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 6]
```diff
@@ -127,64 +106,13 @@ docker-compose up -d
 ----
 
 NOTE: The startup and download can take a while, you can use the `docker-compose ps` command and wait until the health check for the `horizon` service is `up (healthy)`.
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
NOTE: The startup and download can take a while; you can use the `docker-compose ps` command and wait until the health check for the `horizon` service is `up (healthy)`.
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 7]
```diff
@@ -127,64 +106,13 @@ docker-compose up -d
...
-
-=> Everything is awesome
-----
-
-NOTE: The default admin password for the _Minion Karaf Shell_ is _admin_.
+      
+===== Step 3: Login to the Web UI
 
-==== Step 7: Verify status in the administrative Web UI
+After download and startup verify if you can access the web user interface with going to http://localhost:8980.
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
After download and startup verify if you can access the web user interface by going to http://localhost:8980.
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 8]
```diff
@@ -127,64 +106,13 @@ docker-compose up -d
...
+The default login is `admin` with password `admin`.
 
-* Login as admin
-* _Configure OpenNMS -> Manage Minions_, the _Minion_ should be registered and the Status should be _up_
-* Verify if _Minion_ is provisioned automatically going to _Info -> Nodes_ and select the _Minion_, the services _JMX-Minion_, _Minion-Heartbeat_ and _Minion-RPC_ should be _up_ and provisioned on the local loopback interface
+IMPORTANT: Please change immediately your admin account and set a strong password.
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
IMPORTANT: Please immediately change your admin account and set a strong password.
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 9]
```diff
@@ -0,0 +1,103 @@
+
+=== Run with Docker
+
+Modern infrastructure allows you to deploy and run workloads in containers.
+With _{opennms-product-name}_ we provide and publish container images on link:https://hub.docker.com/u/opennms[DockerHub].
+
+==== Objectives
+
+* Run and configure a _Minion_ in and connect it to the _{opennms-product-name}_ instance using environment variables
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
* Run and configure a _Minion_ in, and connect it to, the _{opennms-product-name}_ instance using environment variables
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+* Introduce a reference with all available configuration and mount conventions for more advanced setups
+
+==== Before you begin
+
+It is required you have at least the following components installed:
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
You must have at least the following components installed:
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+
+* Current stable _Docker_ release installed, e.g. installed from link:https://docs.docker.com/[Docker Documentation]
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
* Current stable _Docker_ release installed, e.g., installed from link:https://docs.docker.com/[Docker Documentation]
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+* Current stable _Docker Compose_ installed, e.g. installed from link:https://docs.docker.com/compose/install/[Docker Compose instructions]
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
* Current stable _Docker Compose_ installed, e.g., installed from link:https://docs.docker.com/compose/install/[Docker Compose instructions]
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+* You should have a basic knowledge about _Docker_, _Docker Compose_ with networking, persisting files and mounting directories
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
* You should have a basic knowledge of _Docker_, _Docker Compose_ with networking, persisting files and mounting directories
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+* _{opennms-product-name}_ is configured to accept connections via _ActiveMQ_ and a Minion user with _ROLE_MINION_ exist
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
* _{opennms-product-name}_ is configured to accept connections via _ActiveMQ_ and a Minion user with _ROLE_MINION_ 
```
I think we can remove "exist" unless you mean that the ROLE_MINION must exist?

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+* The _Minion_ can connect to _{opennms-product-name}_ with port `61616/TCP` for _ActiveMQ_ and _REST_ on port `8980/TCP`
+
+==== Quickstart service stack
+
+// No section numbers for step-by-step guide
+:!sectnums:
+
+===== Step 1: Create service stack with a Minion
+
+Create a project directory withg `mkdir opennms-minion` and create inside a `docker-compose.yml` file with the following content:
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
Create a project directory with `mkdir opennms-minion` and create a `docker-compose.yml` file in that directory with the following content:
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+      - OPENNMS_HTTP_URL=http://horizon-instance:8980/opennms<7>
+      - OPENNMS_HTTP_USER=minion-user<8>
+      - OPENNMS_HTTP_PASS=minion-password
+    command: ["-f"]
+    ports:<9>
+      - "8201:8201/tcp"
+      - "162:1162/udp"
+----
+<1> Friendly container name
+<2> Timezone for the Minion
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
<2> Time zone for the _Minion_
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+<3> A defined identifier for this _Minion_, if not set a unique user identifier (_UUID_) will be generated
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
<3> A defined identifier for this _Minion_. If not set, a unique user identifier (_UUID_) will be generated
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+<4> The name of the location the _Minion_ and the connection to the _ActiveMQ_ broker running in _{opennms-product-name}_
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
<4> The name of the location of the _Minion_ and the connection to the _ActiveMQ_ broker running in _{opennms-product-name}_
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+<5> _ActiveMQ_ broker endpoint from _{opennms-product-name}_
+<6> Authentication for ActiveMQ broker
+<7> _REST_ endpoint to to connect to the _{opennms-product-name}_ instance
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
<7> _REST_ endpoint to connect to the _{opennms-product-name}_ instance
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 20]
```diff
@@ -0,0 +1,103 @@
...
+<8> Authentication for the _REST_ endpoint
+<9> Publish ports for SSH access to the _Karaf Shell_ and listen for _SNMP Traps_ forwarding to an internal un-privileged port
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
<9> Publish ports for SSH access to the _Karaf Shell_ and listen for _SNMP Traps_ forwarding to an internal unprivileged port
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 21]
```diff
@@ -0,0 +1,103 @@
...
+
+NOTE: In this example we haven't set credentials to connect the _Minion_ via _REST_ and the _ActiveMQ Message Broker.
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
NOTE: In this example we haven't set credentials to connect the _Minion_ via _REST_ and the _ActiveMQ Message Broker_.
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 22]
```diff
@@ -0,0 +1,103 @@
...
+
+[source,shell]
+----
+cd opennms-minion
+docker-compose up -d
+----
+
+===== Step 3: Run _Minion_ health check
+
+.Login in to the _Minion Karaf Shell_ and run the health check
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
.Log in to the _Minion Karaf Shell_ and run the health check
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 23]
```diff
@@ -0,0 +1,103 @@
...
+Connecting to JMS Broker         [ Success  ]
+
+=> Everything is awesome
+----
+
+NOTE: The default admin password for the _Minion Karaf Shell_ is _admin_.
+
+===== Step 4: Verify status in the administrative Web UI
+
+* Login as admin in the _{opennms-product-name}_ web interface
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
* Log in as admin in the _{opennms-product-name}_ web interface
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 24]
```diff
@@ -0,0 +1,103 @@
...
+* _Configure OpenNMS -> Manage Minions_, the _Minion_ should be registered and the status should be _up_
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
* _Configure OpenNMS -> Manage Minions_. The _Minion_ should be registered and the status should be _up_
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 25]
```diff
@@ -0,0 +1,103 @@
...
+* Verify if _Minion_ is provisioned automatically going to _Info -> Nodes_ and select the _Minion_, the services _JMX-Minion_, _Minion-Heartbeat_ and _Minion-RPC_ should be _up_ and provisioned on the local loopback interface
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
* Verify if _Minion_ is provisioned automatically by going to _Info -> Nodes_ and select the _Minion_. The services _JMX-Minion_, _Minion-Heartbeat_ and _Minion-RPC_ should be _up_ and provisioned on the local loop-back interface
```

**Comment by indigo423** [comment]
Any new things here? Is this mergeable?

**Comment by agalue** [comment]
What about my question about hard-coding versions? On a similar topic, this doesn't make sense for foundation branches (but guess this content is not used for Meridian). 

I'm OK keeping it as "it is", even if that means we have to update the docs on every single version for consistency (if it has to stay hard-coded).

**Comment by Bonrob2** [comment]
Nothing from me.

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 26]
```diff
@@ -0,0 +1,118 @@
...
+
+* Run _{opennms-product-name}_ using _Docker Compose_ with a basic setup and _PostgreSQL_ on your local system as a quickstart
+* Persist RRD files from _{opennms-product-name}_ and _PostgreSQL_ in a volume
+* Introduce a reference with all available configuration and mount conventions for more advanced setups
+
+==== Before you begin
+
+You must have at least the following components installed:
+
+* Current stable _Docker_ release installed, e.g. installed from link:https://docs.docker.com/[Docker Documentation]
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
* Current stable _Docker_ release installed, e.g., installed from link:https://docs.docker.com/[Docker Documentation]
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 27]
```diff
@@ -0,0 +1,118 @@
...
+* Current stable _Docker Compose_ installed, e.g. installed from link:https://docs.docker.com/compose/install/[Docker Compose instructions]
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
* Current stable _Docker Compose_ installed, e.g., installed from link:https://docs.docker.com/compose/install/[Docker Compose instructions]
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 28]
```diff
@@ -0,0 +1,118 @@
...
+* You should have a basic knowledge of _Docker_, _Docker Compose_ with networking, persisting files and mounting directories
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
You should have a basic knowledge of _Docker_, _Docker Compose_ with networking, persisting files and mounting directories
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 29]
```diff
@@ -0,0 +1,118 @@
...
+      - "8980:8980/tcp"
+      - "8101:8101/tcp"
+      - "61616:61616/tcp"
+    healthcheck:<14>
+      test: [ "CMD", "curl", "-f", "-I", "http://localhost:8980/opennms/login.jsp" ]
+      interval: 1m
+      timeout: 5s
+      retries: 3
+----
+<1> Volume definition to persist permanently the _PostgreSQL_ database
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
<1> Volume definition to persist  the _PostgreSQL_ database permanently 
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 30]
```diff
@@ -0,0 +1,118 @@
...
+<2> Volume definition to persist permanently the _RRD_ files from _{opennms-product-name}_
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
<2> Volume definition to persist the _RRD_ files from _{opennms-product-name}_ permanently 
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 31]
```diff
@@ -0,0 +1,118 @@
...
+<3> Service name `database` for the _PostgreSQL_ instance
+<4> Image reference for the vanilla _PostgreSQL_ Docker image with a fixed version
+<5> Friendly container name
+<6> Environment variables to initialize a postgres user with a password.
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
<6> Environment variables to initialize a postgres user with a password
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 32]
```diff
@@ -0,0 +1,118 @@
...
+<7> Assign volume to persist the _PostgreSQL_ database
+<8> Create a health check for the _PostgreSQL_ database
+<9> Image reference for the _{opennms-product-name}_ container image using the latest stable version
+<10> Setup a data base connection using the `postgres` root user and initialize an `opennms` database with user and credentials
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
<10> Set up a database connection using the `postgres` root user and initialize an `opennms` database with user and credentials
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 33]
```diff
@@ -0,0 +1,118 @@
...
+----
+cd opennms-horizon
+docker-compose up -d
+----
+
+NOTE: The startup and download can take a while; you can use the `docker-compose ps` command and wait until the health check for the `horizon` service is `up (healthy)`.
+      
+===== Step 3: Login to the Web UI
+
+After download and startup verify if you can access the web user interface by going to http://localhost:8980.
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
After download and startup, verify that you can access the web user interface by going to http://localhost:8980.
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 34]
```diff
@@ -0,0 +1,118 @@
...
+
+[source,shell]
+----
+cd opennms-horizon
+docker-compose up -d
+----
+
+NOTE: The startup and download can take a while; you can use the `docker-compose ps` command and wait until the health check for the `horizon` service is `up (healthy)`.
+      
+===== Step 3: Login to the Web UI
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
===== Step 3: Log in to the Web UI
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 35]
```diff
@@ -0,0 +1,107 @@
...
+Connecting to OpenNMS ReST API   [ Success  ]
+Verifying installed bundles      [ Success  ]
+Connecting to JMS Broker         [ Success  ]
+
+=> Everything is awesome
+----
+
+NOTE: The default admin password for the _Minion Karaf Shell_ is _admin_.
+
+===== Step 4: Verify status in the administrative Web UI
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
===== Step 4: Verify status in the web UI
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 36]
```diff
@@ -0,0 +1,107 @@
...
+
+===== Step 2: Start the service stack and test the functionality
+
+[source,shell]
+----
+cd opennms-minion
+docker-compose up -d
+----
+
+===== Step 3: Run _Minion_ health check
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
===== Step 3: Run _Minion_ Health Check
```

File: opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc [hunk 37]
```diff
@@ -0,0 +1,118 @@
+
+=== Run with Docker
+
+Modern infrastructure allows you to deploy and run workloads in containers.
+With _{opennms-product-name}_ we provide and publish container images on link:https://hub.docker.com/u/opennms[DockerHub].
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/basic-setup-opennms/docker/opennms-docker.adoc]:
```suggestion
_{opennms-product-name}_ provides and publishes container images on link:https://hub.docker.com/u/opennms[DockerHub].
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 38]
```diff
@@ -0,0 +1,107 @@
+
+=== Run with Docker
+
+Modern infrastructure allows you to deploy and run workloads in containers.
+With _{opennms-product-name}_ we provide and publish container images on link:https://hub.docker.com/u/opennms[DockerHub].
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
_{opennms-product-name}_ provides and publishes container images on link:https://hub.docker.com/u/opennms[DockerHub].
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 39]
```diff
@@ -0,0 +1,107 @@
...
+* Run and configure a _Minion_ in, and connect it to, the _{opennms-product-name}_ instance using environment variables
+* Introduce a reference with all available configuration and mount conventions for more advanced setups
+
+==== Before you begin
+
+You must have at least the following components installed:
+
+* Current stable _Docker_ release installed, e.g., installed from link:https://docs.docker.com/[Docker Documentation]
+* Current stable _Docker Compose_ installed, e.g., installed from link:https://docs.docker.com/compose/install/[Docker Compose instructions]
+* You should have a basic knowledge of _Docker_, _Docker Compose_ with networking, persisting files and mounting directories
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
You should have a basic knowledge of _Docker_, _Docker Compose_ with networking, persisting files and mounting directories
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 40]
```diff
@@ -0,0 +1,107 @@
...
+* _{opennms-product-name}_ is configured to accept connections via _ActiveMQ_ and a Minion user with _ROLE_MINION_
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
_{opennms-product-name}_ is configured to accept connections via _ActiveMQ_ and a Minion user with _ROLE_MINION_ 
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 41]
```diff
@@ -0,0 +1,107 @@
...
+
+==== Before you begin
+
+You must have at least the following components installed:
+
+* Current stable _Docker_ release installed, e.g., installed from link:https://docs.docker.com/[Docker Documentation]
+* Current stable _Docker Compose_ installed, e.g., installed from link:https://docs.docker.com/compose/install/[Docker Compose instructions]
+* You should have a basic knowledge of _Docker_, _Docker Compose_ with networking, persisting files and mounting directories
+* _{opennms-product-name}_ is configured to accept connections via _ActiveMQ_ and a Minion user with _ROLE_MINION_ 
+* The _Minion_ can connect to _{opennms-product-name}_ with port `61616/TCP` for _ActiveMQ_ and _REST_ on port `8980/TCP`
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
The _Minion_ can connect to _{opennms-product-name}_ with port `61616/TCP` for _ActiveMQ_ and _REST_ on port `8980/TCP`
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 42]
```diff
@@ -0,0 +1,107 @@
...
+<3> Time zone for the _Minion_
+<4> A defined identifier for this _Minion_. If not set, a unique user identifier (_UUID_) will be generated
+<5> The name of the location of the _Minion_ and the connection to the _ActiveMQ_ broker running in _{opennms-product-name}_
+<6> _ActiveMQ_ broker endpoint from _{opennms-product-name}_
+<7> Authentication for ActiveMQ broker
+<8> _REST_ endpoint to connect to the _{opennms-product-name}_ instance
+<9> Authentication for the _REST_ endpoint
+
+NOTE: In this example we haven't set credentials to connect the _Minion_ via _REST_ and the _ActiveMQ Message Broker_.
+      The _Minion_ will fall back and uses the default admin/admin credentials for the communication.
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
      The _Minion_ will fall back and uses the default admin/admin credentials for communication.
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 43]
```diff
@@ -0,0 +1,107 @@
...
+      Permissions for _ActiveMQ_ and _REST_ are assigned with the role _ROLE_MINION_ on the _{opennms-product-name}_ instance.
+
+WARNING: In case you process UDP data and you don't use `network_mode: host` the UDP source address from your packets will be modified from Docker.
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
WARNING: If you process UDP data and you don't use `network_mode: host`, the UDP source address from your packets will be modified from Docker.
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 44]
```diff
@@ -0,0 +1,107 @@
...
+         The source address is your Docker internal gateway instead of the source address of your device.
+         Source addresses are used to associate the Syslog or SNMP traps to the nodes in the OpenNMS database.
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
         Source addresses associate the Syslog or SNMP traps to the nodes in the OpenNMS database.
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 45]
```diff
@@ -0,0 +1,107 @@
...
+         You can use an isolated network and publish ports as usual if you don't receive UDP based monitoring data.
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
         You can use an isolated network and publish ports as usual if you don't receive UDP-based monitoring data.
```

File: opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc [hunk 46]
```diff
@@ -0,0 +1,107 @@
...
+=> Everything is awesome
+----
+
+NOTE: The default admin password for the _Minion Karaf Shell_ is _admin_.
+
+===== Step 4: Verify status in the administrative Web UI
+
+* Log in as admin in the _{opennms-product-name}_ web interface
+* _Configure OpenNMS -> Manage Minions_. The _Minion_ should be registered and the status should be _up_
+* Verify if _Minion_ is provisioned automatically by going to _Info -> Nodes_ and select the _Minion_. The services _JMX-Minion_, _Minion-Heartbeat_ and _Minion-RPC_ should be _up_ and provisioned on the local loop-back interface
```
Comment (Bonrob2) [inline opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]:
```suggestion
* Verify that _Minion_ is provisioned automatically by going to _Info -> Nodes_ and selecting the _Minion_. The services _JMX-Minion_, _Minion-Heartbeat_ and _Minion-RPC_ should be _up_ and provisioned on the local loop-back interface
```

Comment (Bonrob2) [reply opennms-doc/guide-install/src/asciidoc/text/monitor-location-minion/docker/minion-docker.adoc]: Could we offer suggestions on what to look for if they are not provisioned? Or is that beyond the scope of this material?

**Comment by indigo423** [comment]
> What about my question about hard-coding versions? On a similar topic, this doesn't make sense for foundation branches (but guess this content is not used for Meridian).
> 
> I'm OK keeping it as "it is", even if that means we have to update the docs on every single version for consistency (if it has to stay hard-coded).

got response from the AsciiDoc community and got it to work :) Thanks for the heads up.

**Pull request merged**